### PR TITLE
Added an ability to override the help message.

### LINKED ISSUE #8: Help
_As a CLI developer, I want to have an automatic `--help` flag for all my interfaces. That means I need to have an ability to specify a description for any argument._

## API
```python
# copy.py

from candies import cli
from candies.cli import Arg

@cli
def main(source: Arg('Source file or directory'),
         destination: Arg('Destination path of a file or a directory')):
    """Script or command description."""
```

So it's possible to call:
```
$ python copy.py --help
Usage: copy.py --source <source> --destination <destination>

    Script or command description.

Arguments:
    --source         Source file or directory
    --destination    Destination path of a file or a directory.    
```

### Subcommands
The syntax of `--help` flag for subcommands looks like:
```
$ python copy.py --help
$ python copy.py COMMAND --help
$ python copy.py COMMAND1 COMMAND2 --help
```

### Arguments before/after `--help`
If some arguments are specified before or after `--help` flag and they're not command names, the invocation of the flag should be considered as invalid:
```
$ python copy.py --source ~/1.txt --help
Invalid invocation of `copy.py`: expected `copy.py --help` or `copy.py --source ~1/.txt ...`.
```

## Requirements
- Code should be tested.

### PR DESCRIPTION
**Pull request by konstantin-ogulchansky**
This PR closes #8.

### CODE REVIEW

File: candies/cli/command.py [hunk 1]
```diff
@@ -70,3 +76,29 @@ def define(x):
...
                define
+
+    def help(self, func: Callable[['Command'], str]) -> Callable:
+        """Defines a function to be called for `--help`.
+
+        The function must be able to accept a command to construct a help
+        message for and return a string with the message to be displayed.
+
+        Args:
+            func: A function to be called when `--help` if specified.
```
Comment (JoshuaLight) [inline candies/cli/command.py:87]: `if` -> `is`?

File: candies/cli/cli.py [hunk 2]
```diff
@@ -44,9 +44,16 @@ def command(self, *args, **kwargs):
         Consider looking at the documentation of the `Command.command` method
         for more details.
         """
-
         return self.main.command(*args, **kwargs)
 
+    def help(self, *args, **kwargs):
+        """Defines a function to be called for `--help`.
+
+        Consider looking at the documentation of the `Command.command` method
```
Comment (JoshuaLight) [inline candies/cli/cli.py:52]: I feel like `Command.command` should be `candies.cli.command.help`.

File: candies/cli/parsers/standard.py [hunk 3]
```diff
@@ -62,7 +60,32 @@ def parse(self, args: Iterable[str]) -> Iterable[Invocation]:
         return invocations
 
 
-def configure(parser: ArgumentParser, command: Command, prefix: str = '.'):
+def help(command: Command) -> Type:
+    """Constructs a subclass of an `argparse.HelpFormatter` for a `command`."""
```
Comment (JoshuaLight) [inline candies/cli/parsers/standard.py:64]: Is `an` and `a` articles are a good fit here?

File: candies/cli/parsers/standard.py [hunk 4]
```diff
@@ -62,7 +60,32 @@ def parse(self, args: Iterable[str]) -> Iterable[Invocation]:
...
+
+    return Help
+
+
+def new_parser(*args, **kwargs) -> ArgumentParser:
+    """Creates a new parser."""
+    return ArgumentParser(*args, **kwargs)
+
+
+def new_subparser(subparsers: Any) -> Callable[[Any], ArgumentParser]:
```
Comment (JoshuaLight) [inline candies/cli/parsers/standard.py:83]: Seems like the return type is missing.

File: candies/cli/parsers/standard.py [hunk 5]
```diff
@@ -62,7 +60,32 @@ def parse(self, args: Iterable[str]) -> Iterable[Invocation]:
...
+    """Returns a function to create a new subparser."""
+    return subparsers.add_parser
+
+
+def configured(new: Callable, command: Command, prefix: str = '.'):
```
Comment (JoshuaLight) [inline candies/cli/parsers/standard.py:88]: And here as well.

File: candies/cli/parsers/standard.py [hunk 6]
```diff
@@ -91,9 +115,9 @@ def configure(parser: ArgumentParser, command: Command, prefix: str = '.'):
         subparsers = parser.add_subparsers(dest=prefix + 'command')
 
         for name, subcommand in command.subcommands.items():
-            subparser = subparsers.add_parser(name)
+            _ = configured(new_subparser(subparsers), subcommand, prefix + '.')
```
Comment (JoshuaLight) [inline candies/cli/parsers/standard.py:118]: Is `_` just to show that the `configured` returns something not needed?

**Comment by JoshuaLight** [comment]
I think that it would be nice to have integration tests for help.

Consider a following example of one:
```py
def test_custom_help():
    # Arrange.
    @cli
    def main():
        ...

    @cli.help
    def help(command):
        return f'Help for `{command.name}`.'

    # Act.
    execute(main, with_='--help')

    # Assert.
    # Check that stdio received 'Help for `main`.'
```

File: candies/cli/command.py (hunk 1 above)
Comment (konstantin-ogulchansky) [reply candies/cli/command.py:87]: Sure. Will fix it.

File: candies/cli/cli.py (hunk 2 above)
Comment (konstantin-ogulchansky) [reply candies/cli/cli.py:52]: Yes.

File: candies/cli/parsers/standard.py (hunk 3 above)
Comment (konstantin-ogulchansky) [reply candies/cli/parsers/standard.py:64]: I don't know, probably not. Will remove them.

File: candies/cli/parsers/standard.py (hunk 5 above)
Comment (konstantin-ogulchansky) [reply candies/cli/parsers/standard.py:88]: Where? There is `Callable[[Any], ArgumentParser]`. Or you mean something else?

Comment (konstantin-ogulchansky) [reply candies/cli/parsers/standard.py:88]: Fixed.

File: candies/cli/parsers/standard.py (hunk 6 above)
Comment (konstantin-ogulchansky) [reply candies/cli/parsers/standard.py:118]: Yes.

File: tests/integration/help_test.py [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+    # Arrange.
+    @cli
+    def main():
+        pass
+
+    # Act.
+    output, error = execute(main, with_='--help')
+
+    # Assert.
+    assert error.code == 0
```
Comment (JoshuaLight) [inline tests/integration/help_test.py:46]: Is it possible to have `error.code != 0` here, if `output.text.startswith('usage: main')` is `True`?

Comment (konstantin-ogulchansky) [reply tests/integration/help_test.py]: Totally removed these checks.

**Pull request merged**